src/chapters/spring-model-classes/data-layer.rst

```diff
@@ -0,0 +1,242 @@
+Models and Data 
+===============
+
+In order to work with data, we need to add another element to our MVC application. Say for example,
+we want to do things like remove an event from our list. Well, if two events both have the same name, 
+how might we identify which of those items to delete? We can't yet. So we need to tweak how we store
+event data. 
+
+In ``coding-events``, we add a unique identifier field to ``Events`` to better handle and track distinct 
+``Event`` instances. We'll also create another model class called ``EventData`` to encapsulate data storage 
+and prepare ourselves for decoupling data from model classes.
+
+Add a Unique Id - Video
+-----------------------
+
+.. todo:: video adding a unique id generator
+
+.. index:: ! uid
+
+Add a Unique Id - Text
+-----------------------
+
+Identifying data by a user-defined string called ``name`` is not a sustainable or scalable method
+of handling data in most situations. Consider the address book example. How can
+we distinguish between two contact entries with the same name field? It is a frequent
+practice to add a **unique identifier** field (sometimes called, or even labelled, **uid**) to a class 
+responsible for modelling data. This ensures that our address book can contain two separate entries for 
+our contacts who have the same name as one another. 
+
+To accomplish the same data clarity with events, we'll add a few things to the event model class:
+
+#. A private ``id`` field .
+#. A static counter variable, ``nextId``.
+#. A no-arg constructor that:
+   
+   a. Sets the ``id`` field to the ``nextId`` value.
+   b. Increments ``nextId``.
+
+#. A call to the default constructor in all other constructors.
```

Comment:
"default constructor" has a slightly different, precise meaning, so we should use "no-arg constructor" in situations like this.